### DISCUSSION TITLE
Question about nolibfuse.

### DISCUSSION THREAD

**Discussion by Samueru-sama** [ivan-hc/AM#387] [Q&A] [answered]
If I remove the fuse2 package, the only appimage that works on my system is amdgpu_top.

I tested the nolibfuse option in AM to turn some type 2 appimages into type 3. those do not work out of the box, however I don't get the fuse2 error, I instead get a missing fusermount error. 

however if I make a symlink to fusermount3, then the issue goes away and those appimages work without fuse2: 

![image](https://github.com/ivan-hc/AM/assets/36420837/7be50cf5-c1ba-48a9-aa3d-f4e9efb5e370)

Is this a bug? or is the distro expected to symlink fusermount to fusermount3?

**Comment by ivan-hc** [comment]
You shoul ask this to Probonopd. Actually it still depends on the newer version of FUSE, but no more on the version 2 of libfuse, for security reasons

**Reply by Samueru-sama** [reply]
This also happens with the appimages that I made using go-appimage. (deadbeef, ryujinx, android-tools), they can only work with fuse3 if I symlink fusermount to fusermount3.

@probonopd

**Reply by Samueru-sama** [reply]
Btw @ivan-hc what happens on your system if you remove fuse2 but still have fuse3 installed? do your type 3 appimages work in your system? or you have to do the same symlink as me for them to work?

**Reply by ivan-hc** [reply]
I don't use libfuse2 anymore

**Reply by Samueru-sama** [reply]
@ivan-hc can I see the output of `ls /usr/bin | grep fusermount` ?

**Reply by ivan-hc** [reply]
Sure, I've fusermount3 (Debian)
```
fusermount
fusermount3
```

**Reply by Samueru-sama** [reply]
You have fuse2 though, that is what fusermount is. Now is `fusermount` a symlink to fusermount3 or just its own binary?

**Reply by ivan-hc** [reply]
Nope, I'm on Debian, this is the package https://packages.debian.org/trixie/fuse3 and this is its content:
```
/.
/bin
/bin/fusermount
/bin/fusermount3
/etc
/etc/fuse.conf
/sbin
/sbin/mount.fuse
/sbin/mount.fuse3
/usr
/usr/share
/usr/share/doc
/usr/share/doc/fuse3
/usr/share/doc/fuse3/changelog.Debian.gz
/usr/share/doc/fuse3/changelog.gz
/usr/share/doc/fuse3/copyright
/usr/share/initramfs-tools
/usr/share/initramfs-tools/hooks
/usr/share/initramfs-tools/hooks/fuse
/usr/share/man
/usr/share/man/man1
/usr/share/man/man1/fusermount.1.gz
/usr/share/man/man1/fusermount3.1.gz
/usr/share/man/man8
/usr/share/man/man8/mount.fuse.8.gz
/usr/share/man/man8/mount.fuse3.8.gz
```

**Reply by ivan-hc** [reply]
and yes, fusermount is just a symlink to fusermount3

**Reply by ivan-hc** [reply]
wow, just seen that on Arch Linux, Fusermount is into a separated package, https://archlinux.org/packages/extra/x86_64/fuse2

**Reply by Samueru-sama** [reply]
@ivan-hc Yeah in arch `fusermount` is fuse2 and `fusermount3` is fuse3, there is no symlink of fusermount to fusermount3 like in debian, so if only fusermount3 is installed the appimages wont work out of the box. @probonopd

**Reply by ivan-hc** [reply]
a symlink or a script that allows fusermount to be used as fusermount or fusermount3 would be needed, the way "firefox" switches between firefox and firefox-esr, in Debian

**Reply by Samueru-sama** [reply]
@ivan-hc probono just fixed it! So no hacks will be needed lol.

**Comment by probonopd** [comment]
Which exact command are you using to make the symlink?

This code _should_ address this issue:

https://github.com/AppImage/type2-runtime/blob/9594e2b1306a83b1020145f4e4e29c61065536a4/patches/libfuse/mount.c.diff#L59-L71

**Reply by Samueru-sama** [reply]
@probonopd

I'm using `ln -s` to make the symlink of fusermount that points to fusermount3, it is the only way that the appimage works when fuse2 is not installed.

Also this issue is present in my own appimages that I have in my pinned repos, example: https://github.com/Samueru-sama/DeaDBeef-AppImage

![image](https://github.com/ivan-hc/AM/assets/36420837/fd8f5027-bf62-497d-ae6e-5d8871733137)

I remove fuse2 and try to start deadbeef, I get fusermount missing error. Then I `ln -s /usr/bin/fusermount3 /usr/bin/fusermount` and the app starts without problems afterwards. 

(This on archlinux btw).

**Reply by probonopd** [reply]
Which exact tool do you use to make your AppImages?

**Reply by ivan-hc** [reply]
@probonopd him is using https://github.com/probonopd/go-appimage/releases/download/continuous/appimagetool-823-x86_64.AppImage , I suggested it to him

EDIT: I use it in all my workflows and never had issues

**Reply by Samueru-sama** [reply]
@probonopd go-appimage appimagetool. 

In the `deadbeef-appimage.sh` script it is lines 32 to the end of the script for how the appimage is made.

**Comment by Samueru-sama** [comment]
@ivan-hc @probonopd Just opened an issue about that: https://github.com/probonopd/go-appimage/issues/278

**Reply by probonopd** [reply]
Thanks, I am on it.

**Reply by probonopd** [reply, marked as answer]
Done :)
